### PR TITLE
qa/tasks/cephadm.py: Support already bootstrapped clusters

### PR DESCRIPTION
**Pull request by gekios**
Add the possibility of being able to run cephadm.py on clusters that
were already bootstrapped by some other task/way so you can deploy the
rest of the daemons/services.

Signed-off-by: Georgios Kyratsas <gkyratsas@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/tasks/cephadm.py [hunk 1]
```diff
@@ -981,105 +985,110 @@ def task(ctx, config):
...
         config['cluster'] = 'ceph'
     cluster_name = config['cluster']
-    ctx.ceph[cluster_name] = argparse.Namespace()
-
-    ctx.ceph[cluster_name].thrashers = []
-    # fixme: setup watchdog, ala ceph.py
-
-    ctx.ceph[cluster_name].roleless = False  # see below
+    # Support already bootstrapped clusters
+    if not hasattr(ctx.ceph[cluster_name], 'bootstrapped'):
```
Comment (sebastian-philipp) [inline qa/tasks/cephadm.py:989]: can we move all of the code below here into a dedicated method?

Comment (gekios) [reply qa/tasks/cephadm.py]: Done

**Comment by sebastian-philipp** [comment]
```
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/run_tasks.py", line 90, in run_tasks
    manager.__enter__()
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-swagner-testing-2020-05-06-1716/qa/tasks/cephadm.py", line 1105, in task
    lambda: ceph_clients(ctx=ctx, config=config),
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/contextutil.py", line 56, in nested
    reraise(*exc)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/virtualenv/lib/python3.6/site-packages/six.py", line 703, in reraise
    raise value
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/contextutil.py", line 29, in nested
    mgr = mgr_fn()
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-swagner-testing-2020-05-06-1716/qa/tasks/cephadm.py", line 1085, in <lambda>
    if not (ctx.ceph[cluster_name].bootstrapped) else _pass(),
KeyError: 'ceph'
```
http://pulpito.ceph.com/swagner-2020-05-07_08:46:28-rados-wip-swagner-testing-2020-05-06-1716-distro-basic-smithi/

**Comment by gekios** [comment]
I had mistakenly put too much code in the function so I moved the namespace instantiation back in the task. Hopefully it should be fine now

**Comment by sebastian-philipp** [comment]
http://pulpito.ceph.com:80/swagner-2020-05-08_09:28:04-rados-wip-swagner3-testing-2020-05-07-1813-distro-basic-smithi/

**Comment by gekios** [comment]
Oh I had missed this line `cluster_name = config['cluster']` in the function... fyi it's public holiday here today so I will be back on Monday

**Comment by sebastian-philipp** [comment]
>  fyi it's public holiday here today so I will be back on Monday

enjoy!

**Comment by sebastian-philipp** [comment]
http://pulpito.ceph.com/swagner-2020-05-11_17:31:17-rados-wip-swagner2-testing-2020-05-11-1650-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
jenkins test make check

**Comment by sebastian-philipp** [comment]
jenkins test make check

**Comment by gekios** [comment]
I triggered a run and it looked good apart for 2 tests were seemingly running for like 7 hours and after checking most likely there was some malfunction since they weren't locked. http://pulpito.ceph.com/gkyratsas-2020-05-14_15:43:26-rados:cephadm-wip-gkyratsas-testing-34900-distro-basic-smithi/

**Comment by gekios** [comment]
I investigated a bit and found that the test that's failing to me seems broken in general for `ubuntu_latest`. http://pulpito.ceph.com/gkyratsas-2020-05-15_14:21:35-rados:cephadm-wip-gkyratsas-testing-34900-distro-basic-smithi/   is my run compared with the last run I found on top of master branch http://pulpito.ceph.com/nojha-2020-05-12_16:52:34-rados:cephadm-master-distro-basic-smithi/

**Comment by gekios** [comment]
I have triggered a test on top of master filtering out `ubuntu_latest` and it was successful. http://pulpito.ceph.com/gkyratsas-2020-05-18_10:48:02-rados:cephadm-master-distro-basic-smithi/

**Pull request merged**